Calibration status

mkuja opened the pull request:
- Sensor reports come with calibration status [0-3] with 3 being fully calibrated and 0 uncalibrated.
- Support for using sensorhub interrupts.
- Flattened ceva sh2 driver submodule to be part of this repo.
- Add storing calibration data

mkuja committed:
Adding Three -- Non working state

mkuja committed:
Add calibration status to sensor reports

Also change package.json

mkuja committed:
Add interrupts

mkuja committed:
Make use of the interrupt timestamp

mkuja committed:
Convert to libgpiod2, wip

mkuja committed:
Use libgpiod 2

mkuja committed:
Added interrupts

mkuja committed:
Store dynamic calibration data FRS record

Added convenience function for to store dynamic calibration data and
fixed the setFrs function a bit.

mkuja committed:
Add interrupts and debug printing for seq nums

mkuja committed:
Clean exit when using interrupts

mkuja committed:
Change from prepare script to preinstall

mkuja committed:
Flatten sh2 driver submodule

mkuja committed:
Remove submodule scripts from package.json

jtholmi commented on src/example.ts:
```diff
@@ -38,25 +38,23 @@ async function main(): Promise<void> {
     bindings.setSensorConfig(SensorId.SH2_MAGNETIC_FIELD_UNCALIBRATED, OFF)
     bindings.setSensorConfig(SensorId.SH2_RAW_MAGNETOMETER, OFF)
     bindings.setSensorConfig(SensorId.SH2_ROTATION_VECTOR, ON)
+    bindings.useInterrupts("gpiochip0", 23) // Use GPIO23 for interrupts
```
What's difference between this and example_with_interrupts.ts? This looks like you started here and then moved to new file but forgot to restore this.

jtholmi commented on src/example.ts:
```diff
@@ -38,25 +38,23 @@ async function main(): Promise<void> {
-        await sleep(100)
-        bindings.service()
-    }
+    await sleep(5000)
```
I'd restore the original lines using `git checkout wip -p -- src/example.ts` and then selecting the lines on the left. Related to comment https://github.com/ExpeditionExploration/openi2c-BNO08x/pull/1#discussion_r2324721576

mkuja committed:
Remove node path from .clangd

```bash
curl -o- https://fnm.vercel.app/install | bash
fnm install 22

export CFLAGS+="$CFLAGS -I$HOME/.local/share/fnm/node-versions/v22.18.0/installation/include/"
export PATH="$PATH:$HOME/.local/bin"
```

jtholmi commented on src/c-src/funcs.c:
```diff
@@ -701,3 +716,120 @@ napi_value cb_getFrs(napi_env env, napi_callback_info info) {
+        return NULL;
+    }
+
+    uint32_t data[300];
```
Shouldn't be uint8_t based on comment below?

jtholmi commented on src/c-src/funcs.c:
```diff
@@ -701,3 +716,120 @@ napi_value cb_getFrs(napi_env env, napi_callback_info info) {
+                         "Couldn't get current dynamic calibration.");
+        return NULL;
+    }
+    code = sh2_setFrs(DYNAMIC_CALIBRATION, data, words);
```
We need to comment here why we need to get (supposedly from volatile memory) and set (supposedly to nonvolatile memory, e.g., flash)

mkuja committed:
Plenty of small things

- Check some error codes at interrupt.c
- Remove CPU affinity and priority change
- Fix dynamic calibration data storing function
- Fix example and example_with_interrups to do what the naming says
